satpy/tests/test_scene.py

```diff
@@ -120,6 +120,20 @@ def test_init_with_empty_filenames(self):
         filenames = []
         Scene(filenames=filenames)
 
+    def test_init_with_fsfile(self):
+        """Test initialisation with FSFile objects."""
+        from satpy.readers import FSFile
+
+        # We should not mock _create_reader_instances here, because in
+        # https://github.com/pytroll/satpy/issues/1605 satpy fails with
+        # TypeError within that method if passed an FSFile instance.
+        # Instead rely on the ValueError that satpy raises if no readers
+        # are found.
+
+        fsf = FSFile("dummy")
```

Comment:
I think a unique name should be created here. There could be situation where a `dummy` file exists in the directory. Maybe use `mkstemp`?

Reply:
@gerritholl any opinion on this?

Reply:
Oops, I had overlooked this one.  `mkstemp` actually creates a temporary file, which we don't need here.  It doesn't look like the `tempfile` module includes routines to create a temporary filename that isn't a temporary file.  I believe that even if a file "dummy" exists satpy is still going to say "no supported readers found", because no file pattern will match... but I've replaced it by a random string now (without tempfile).